### ISSUE TITLE
The module could not use in new version of Openresty.

### ISSUE THREAD

**Issue by caichudong** [openresty/stream-lua-nginx-module#77]
in openresty v.1.9.* can build and use stream and ngx_lua code.(use the v.0.0.1 stream-lua-nginx-module which u give the link to get this.)

in openresty v.1.11.* can not use stream ,when I test the nginx conf,it echo "段错误".(use v.0.0.1 or master).
![image](https://user-images.githubusercontent.com/26949885/31540595-af2258f8-b03e-11e7-8603-b581f6f8d461.png)
![image](https://user-images.githubusercontent.com/26949885/31540641-c9e54434-b03e-11e7-80fb-dc5f574b34ea.png)

**Comment by agentzh** [comment]
@caichudong Yes, the latest ngx_stream_lua_module requires at least the NGINX 1.13.4 core to work properly while the current OpenResty release is 1.11.2.5 which is based on NGINX 1.11.2.

The plan is to release a new OpenResty version with a new 1.13.x NGINX core (and also the ngx_stream_lua_module) this month or so.

**Comment by caichudong** [comment]
@agentzh Thank u, now I had used the old version to solve this problem.I'm looking forward for the new version.

**Comment by idevz** [comment]
@caichudong 
you can use [openresty-1.11.2.5.tar.gz](https://openresty.org/download/openresty-1.11.2.5.tar.gz), and use the newest ngx_stream_lua_module in the master branch , then use [nginx-1.13.6.tar.gz](http://nginx.org/download/nginx-1.13.6.tar.gz) to instead of the one in the openresty bundle, I did this, and it works for me. 
we can use this way to try the new feature of ngx_stream_lua_module. And use the stard version once its release.

**Comment by caichudong** [comment]
@idevz ok, thanks for ur advice. I'll try it.

**Comment by agentzh** [comment]
@idevz Better avoid replacing nginx core in the openresty bundle since it may break compatibility. Better use the following OpenResty preview version instead:

http://openresty.org/download/openresty-1.13.5.1rc0.tar.gz

**Comment by idevz** [comment]
@agentzh thanks a lot, I just want to try the newest ngx_stream_lua_module for test, so didn't think much more. thanks for your opinion

**Comment by idevz** [comment]
@caichudong you can use this preview version instead too.
http://openresty.org/download/openresty-1.13.5.1rc0.tar.gz

**Comment by idevz** [comment]
when I install openresty-1.13.5.1rc0.tar.gz , I got this:

```bash
cd /media/psf/g/idevz/code/src/openresty-1.13.5.1rc0/../pcre-8.39 \
&& gmake libpcre.la
gmake[3]: 进入目录“/media/psf/g/idevz/code/src/pcre-8.39”
 cd . && /bin/sh /media/psf/g/idevz/code/src/pcre-8.39/missing automake-1.13.4 --gnu
/media/psf/g/idevz/code/src/pcre-8.39/missing:行81: automake-1.13.4: 未找到命令
WARNING: 'automake-1.13.4' is missing on your system.
         You should only need it if you modified 'Makefile.am' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'automake' program is part of the GNU Automake package:
         <http://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <http://www.gnu.org/software/autoconf>
         <http://www.gnu.org/software/m4/>
         <http://www.perl.org/>
gmake[3]: *** [Makefile.in] 错误 1
gmake[3]: 离开目录“/media/psf/g/idevz/code/src/pcre-8.39”
gmake[2]: *** [/media/psf/g/idevz/code/src/openresty-1.13.5.1rc0/../pcre-8.39/.libs/libpcre.a] 错误 2
gmake[2]: 离开目录“/media/psf/g/idevz/code/src/openresty-1.13.5.1rc0/build/nginx-1.13.5”
gmake[1]: *** [build] 错误 2
gmake[1]: 离开目录“/media/psf/g/idevz/code/src/openresty-1.13.5.1rc0/build/nginx-1.13.5”
gmake: *** [all] 错误 2
```

and then, I update the Makefile generate under the pcre-8.39 dir, remove the version info from

```bash
ACLOCAL = ${SHELL} /media/psf/g/idevz/code/src/pcre-8.39/missing aclocal-1.13.4
AMTAR = $${TAR-tar}
AM_DEFAULT_VERBOSITY = 0
AR = ar
AS = as
AUTOCONF = ${SHELL} /media/psf/g/idevz/code/src/pcre-8.39/missing autoconf
AUTOHEADER = ${SHELL} /media/psf/g/idevz/code/src/pcre-8.39/missing autoheader
AUTOMAKE = ${SHELL} /media/psf/g/idevz/code/src/pcre-8.39/missing automake-1.13.4
```
to

```bash
ACLOCAL = ${SHELL} /media/psf/g/idevz/code/src/pcre-8.39/missing aclocal
AMTAR = $${TAR-tar}
AM_DEFAULT_VERBOSITY = 0
AR = ar
AS = as
AUTOCONF = ${SHELL} /media/psf/g/idevz/code/src/pcre-8.39/missing autoconf
AUTOHEADER = ${SHELL} /media/psf/g/idevz/code/src/pcre-8.39/missing autoheader
AUTOMAKE = ${SHELL} /media/psf/g/idevz/code/src/pcre-8.39/missing automake
```
and It's work for me.

so, @agentzh this is because of the pcre version or any other reason? I found that the pcre use in the [new OpenResty Installation](https://github.com/openresty/openresty.org/blob/master/v2/en/installation.md#make-install) is using pcre-8.40, my net access was forbidden, and I can't get pcre-8.40 from ftp.csx.cam.ac.uk for try this :(.

**Comment by agentzh** [comment]
@idevz It's strange that it triggers automake in the pcre repo. This should never happen. I wonder if your system's clock is screwed up?

**Comment by idevz** [comment]
@agentzh may be you're right, I did this in my centos VM, which clock is always more later then the host mac. Thanks a lot.

**Comment by agentzh** [comment]
OK, a new openresty preview version is provided for testing:

https://openresty.org/download/openresty-1.13.6.1rc0.tar.gz

It includes the latest nginx 1.13.6 core.

**Comment by sparvu** [comment]
10 x thanks

**Comment by agentzh** [comment]
BTW, this openresty 1.13.6.1 rc0 preview version already includes and enables ngx_stream_lua_module by default.

**Comment by idevz** [comment]
Using openresty 1.13.6.1 rc0 to handle motan(weibo RPC framework) request in weibo, we can do many common things such as validation, filter and so on at preread phase through preread_by_lua_file, its so awsome and cool. Thanks @agentzh, @dndx

**Comment by agentzh** [comment]
@idevz Good to know!

**Comment by henriquechehad** [comment]
when is planning to release stable openresty with nginx 1.13.* ?

**Comment by agentzh** [comment]
@henriquechehad OpenResty 1.13.6.1 RC1 will be out in the next couple of days.

**Comment by dndx** [comment]
@agentzh has recently released OpenResty 1.13.6.1 RC1:

https://groups.google.com/forum/#!topic/openresty-en/pVPxLB5jfl4

**Comment by agentzh** [comment]
Consider it resolved.
